src/FullyQualifiedName.elm

```diff
@@ -61,8 +77,21 @@ fromList segments_ =
 fromUrlString : String -> FQN
 fromUrlString str =
     str
-        |> String.replace "/" "."
-        |> fromString
+        |> String.split "/"
+        |> fromUrlList
+
+
+fromUrlList : List String -> FQN
+fromUrlList segments_ =
+    let
+        urlDecode s =
+            -- Let invalid % encoding fall through, since it then must be valid
+            -- strings
+            Maybe.withDefault s (Url.percentDecode s)
```

Comment:
`percentDecode` seems a bit strange to me in that it fails if it encounters a % not followed by valid encoding characters. That kind of configuration are still valid URLs though.